src/Surfnet/AzureMfa/Application/Service/AzureMfaService.php

```diff
@@ -157,9 +157,12 @@ public function createAuthnRequest(User $user, bool $forceAuthn = false): string
             $forceAuthn
         );
 
-        // Use email address as subject
-        $this->logger->info('Setting the users email address as the Subject');
-        $authnRequest->setSubject($user->getEmailAddress()->getEmailAddress());
+        // Use email address as subject if not sending to an AzureAD IdP
+        // TODO: This should be a config parameter in institutions.yaml instead of string matching
+        if (!(strpos($destination->getUrl(), "https://login.microsoftonline.com/") === 0)) {
```

Comment:
I'd like to see this destination url be configurable. Urls tend to change over time, not having to change the code is a great advantage in my opinion. If you don't feel that is valuable enough, then please turn this Url in a class constant.

Reply:
Thank you for the suggestion. Ideally the institution configuration will hold a boolean indicating it is an AzureAD of ADFS. For now I've moved the url to a class constant, for the login-url for all microsoft services is not likely to change often.